sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -57,23 +56,33 @@ import org.apache.spark.sql.util.SchemaUtils
  * The CatalogTable attributes copied from the source table are storage(inputFormat, outputFormat,
  * serde, compressed, properties), schema, provider, partitionColumnNames, bucketSpec.
  *
+ * Use "CREATE TABLE t1 LIKE t2 USING file_format"
+ * to specify new file format for t1 from a data source table t2.
+ *
  * The syntax of using this command in SQL is:
  * {{{
  *   CREATE TABLE [IF NOT EXISTS] [db_name.]table_name
- *   LIKE [other_db_name.]existing_table_name [locationSpec]
+ *   LIKE [other_db_name.]existing_table_name [USING provider] [locationSpec]
  * }}}
  */
 case class CreateTableLikeCommand(
     targetTable: TableIdentifier,
     sourceTable: TableIdentifier,
     location: Option[String],
-    ifNotExists: Boolean) extends RunnableCommand {
+    ifNotExists: Boolean,
+    provider: Option[String]) extends RunnableCommand {
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
     val sourceTableDesc = catalog.getTempViewOrPermanentTableMetadata(sourceTable)
 
-    val newProvider = if (sourceTableDesc.tableType == CatalogTableType.VIEW) {
+    val newProvider = if (provider.isDefined) {
+      val providerWithLowerCase = provider.get.toLowerCase(Locale.ROOT)
+      // check the validation of provider input, invalid provider will throw
+      // AnalysisException or ClassNotFoundException or NoSuchMethodException
+      DataSource.lookupDataSource(providerWithLowerCase, sparkSession.sessionState.conf)
```

Comment:
We already have any test for the error cases in create table like?

Reply:
This is for general verification like ClassNotFoundException if the sql is "create table .. like ... using unknown"

Reply:
I meant "could you add tests for the error cases in some test suites?"

Reply:
Of course. I will add tests.

Reply:
Thanks!

Reply:
@maropu please help to review the test:
https://github.com/apache/spark/blob/82d5a7310d1447abbe1c10947fde9fc8bc2acaea/sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala#L2821